frontend/src/services/web3-provider/coinbase-provider.ts

```diff
@@ -56,11 +56,16 @@ export async function createCoinbaseProvider(rpcList: {
   return undefined;
 }
 
-export class CoinbaseProvider extends EthereumProvider {
-  constructor(_provider: Eip1193Provider) {
+export class CoinbaseProvider extends EthereumProvider<CoinbaseWalletProvider> {
+  constructor(_provider: CoinbaseWalletProvider) {
     super(_provider);
   }
 
+  async closeExternalConnection() {
+    this.externalProvider.close();
+    this.externalProvider.disconnect();
```

Comment:
These methods are actually not async. 

The reason why this function is declared as async is because the function signature inside `ethereum-provider.ts` dictates the async type.